core/src/main/java/io/questdb/cairo/sql/RecordCursor.java

```diff
@@ -36,6 +36,25 @@
  */
 public interface RecordCursor extends Closeable, SymbolTableSource {
 
+    /**
+     * Counts remaining number of records in this cursor.
+     */
+    default long calculateSize(SqlExecutionCircuitBreaker circuitBreaker) {
```

Comment:
this method deserves a more comprehensive javadoc.

Doc should mention that this method mutates the state of the cursor, making it effectively skip to the end of the cursor. It also cannot be called in succession, unless cursor.toTop() is called. To that end, we need tests in `assertQuery()` to ensure calculateSize() produces the same value if cursor is re-opened and after toTop() is called. Perhaps calculate size should throw exception if cursor is already exhausted?

Reply:
I think that to make cold storage really work with this method and skipTo() we'll need to either: 
- add isSkipped flag and rowsAlreadySkipped counter to all factories (otherwise some combinations  of factories might skip more rows than necessary on DataUnavailableException because the default skipped counter is lost) 
or
- pass an external object to store the number of rows frequently (e.g. long[1] )